lib/SPIRV/SPIRVUtil.cpp

```diff
@@ -1254,8 +1254,16 @@ Value *getScalarOrArrayConstantInt(Instruction *Pos, Type *T, unsigned Len,
     assert(Len == 1 && "Invalid length");
     return ConstantInt::get(IT, V, IsSigned);
   }
-  if (auto PT = dyn_cast<PointerType>(T)) {
-    auto ET = PT->getPointerElementType();
+  if (isa<PointerType>(T)) {
+    unsigned PointerSize = Pos->getParent()
+                               ->getParent()
+                               ->getParent()
+                               ->getDataLayout()
+                               .getPointerTypeSizeInBits(T);
+    auto *ET = Type::getIntNTy(T->getContext(), PointerSize);
+    assert(
+        (T->isOpaquePointerTy() || T->getNonOpaquePointerElementType() == ET) &&
+        "Only call this method with pointer-to-size_t arguments");
```

Comment:
Technically, not the most correct assert message, since the very first line of the function is about processing integers.
`if (auto IT = dyn_cast<IntegerType>(T)) {`